### PR TITLE
Split imports binder into its own binder.

### PR DESCRIPTION
**Pull request by AlekseyTs**
This is a refactoring, no behavior change intended.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review.

**Review by cston — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by 333fred** [comment]
Looking.

File: src/Compilers/CSharp/Portable/Binder/ExtensionMethodScope.cs [hunk 1]
```diff
@@ -70,17 +68,8 @@ public bool MoveNext()
...
-                    // for the namespace, this time exposed for the usings.
-                    _current = new ExtensionMethodScope(binder, searchUsingsNotNamespace: true);
-                }
-                else
-                {
-                    // Return a scope for the next Binder that supports extension methods.
-                    _current = GetNextScope(binder.Next);
-                }
+                // Return a scope for the next Binder that supports extension methods.
+                _current = GetNextScope(binder.Next);
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Binder/ExtensionMethodScope.cs:72]: Nit: it looks like this whole if/else could be simplified to `_current = GetNextScope(_current.Binder?.Next ?? _binder);`

**Review by 333fred — APPROVED** [review]
LGTM (commit 2). One suggested refactoring, but that's fine as a followup.

**Pull request merged**